src/Nitrate/Optimizations/ParticleRendering/Dust/InstancedDustRenderer.cs

```diff
@@ -113,14 +116,74 @@ internal sealed class InstancedDustRenderer : AbstractInstancedParticleRenderer<
+
+            //FasterParallel breaks it
+            if (dust.active && dust.type >= DustID.Count) {
+                if (new Rectangle((int)dust.position.X, (int)dust.position.Y, 4, 4).Intersects(rectangle)) {
+                    float scale = dust.GetVisualScale();
+                    if (dust.shader != armorShaderData) {
+                        Main.spriteBatch.End();
+                        armorShaderData = dust.shader;
+                        if (armorShaderData == null) {
+                            Main.spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend,
+                                SamplerState.PointClamp, DepthStencilState.None, RasterizerState.CullNone, null,
+                                Main.Transform);
+                        } else {
+                            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend,
+                                SamplerState.PointClamp, DepthStencilState.None, RasterizerState.CullNone, null,
+                                Main.Transform);
+                            dust.shader.Apply(null);
+                        }
+                    }
+
+                    Color newColor = Lighting.GetColor((int)(dust.position.X + 4.0) / 16, (int)(dust.position.Y + 4.0) / 16);
+                    Color alpha = dust.GetAlpha(newColor);
+                    ModDust dust2 = DustLoader.GetDust(dust.type);
+                    if (dust2 != null) {
+                        if (dust2.PreDraw(dust)) {
+                            dust2.Draw(dust, alpha, scale);
+                        }
+
+                        if (alpha == Color.Black) {
+                            dust.active = false;
+                        }
+                    } else {
+                        Main.spriteBatch.Draw(TextureAssets.Dust.Value, dust.position - Main.screenPosition, new Rectangle?(dust.frame), alpha, dust.GetVisualRotation(), new Vector2(4f, 4f), scale, SpriteEffects.None, 0.0f);
+                        if (dust.color.PackedValue != 0U) {
+                            Color color = dust.GetColor(alpha);
+                            if (color.PackedValue != 0U) {
+                                Main.spriteBatch.Draw(TextureAssets.Dust.Value,
+                                    dust.position - Main.screenPosition,
+                                    new Rectangle?(dust.frame), color,
+                                    dust.GetVisualRotation(), new Vector2(4f, 4f), scale, SpriteEffects.None,
+                                    0.0f);
+                            }
+                        }
+
+                        if (alpha == Color.Black) {
+                            dust.active = false;
+                        }
+                    }
+                } else {
+                    dust.active = false;
+                }
+            }
+        }
+
+        Main.spriteBatch.End();
+        Main.pixelShader.CurrentTechnique.Passes[0].Apply();
```

Comment:
Is this directly copied from the dust rendering code? I'd rather we just IL edit the vanilla rendering code and allow it to run instead, limiting it only to modded dusts.